scripts/styleChecker.fsx

```diff
@@ -0,0 +1,393 @@
+    match processResult.Result with
+    | Success output -> true
+    | _ -> false
+
+let InstallPrettier(version: string) =
+    let isPrettierInstalled =
+        Process.Execute(
+            {
+                Command = "npm"
+                Arguments = $"list prettier@{version}"
+            },
+            Echo.All
+        )
+        |> IsProcessSuccessful
+
+    if not(isPrettierInstalled) then
+        Process.Execute(
+            {
+                Command = "npm"
+                Arguments = $"install prettier@{version}"
+            },
+            Echo.Off
+        )
+        |> UnwrapProcessResult "" true
+        |> ignore
+
+let StyleFSharpFiles(rootDir: DirectoryInfo) =
+    InstallFantomlessTool(fantomlessToolVersion)
+
+    Process
+        .Execute(
+            {
+                Command = "dotnet"
+                Arguments = $"fantomless --recurse {rootDir.FullName}"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let StyleCSharpFiles(rootDir: DirectoryInfo) =
+    Process
+        .Execute(
+            {
+                Command = "dotnet"
+                Arguments = $"format whitespace {rootDir.FullName} --folder"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let InstallPrettierPluginXml(version: string) =
+    let isPrettierPluginXmlInstalled =
+        Process.Execute(
+            {
+                Command = "npm"
+                Arguments = $"list @prettier/plugin-xml@{version}"
+            },
+            Echo.Off
+        )
+        |> IsProcessSuccessful
+
+    if not(isPrettierPluginXmlInstalled) then
+        Process
+            .Execute(
+                {
+                    Command = "npm"
+                    Arguments = $"install @prettier/plugin-xml@{version}"
+                },
+                Echo.Off
+            )
+            .UnwrapDefault()
+        |> ignore
+
+let StyleXamlFiles() =
+    InstallPrettier(prettierVersion)
+    InstallPrettierPluginXml(pluginXmlVersion)
+
+    Process
+        .Execute(
+            {
+                Command = "npm"
+                Arguments =
+                    $"install --save-dev prettier@{prettierVersion} @prettier/plugin-xml@{pluginXmlVersion}"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+    Process
+        .Execute(
+            {
+                Command = "./node_modules/.bin/prettier"
+                Arguments =
+                    "--xml-whitespace-sensitivity ignore --tab-width 4 --prose-wrap preserve --write '**/*.xaml'"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let RunPrettier(arguments: string) =
+
+    // We need this step so we can change the files using `npx prettier --write` in the next step.
+    // Otherwise we get permission denied error in the CI.
+    Process
+        .Execute(
+            {
+                Command = "chmod"
```

Comment:
@realmarv let's only do this in case environment variable GITHUB_REF is not empty (to make sure this is only done when in CI; take in account that we may ask our teammates to run this script soon from their local environments, not just CI)

Reply:
If we want to run the script outside CI we first need to make sure there are no uncommitted changes in the local machine

Reply:
why? when a developer runs fantomas or fantomless on his repo, fantomas/fantomless doesn't check git at all; and this .fsx script is just trying to do a similar thing as fantomless, but adjusting style of more files, not just F#